tests/integration/test_api_keys_routes.py

```
@@ -38,6 +38,244 @@ def test_save_api_keys_and_read_back(client, monkeypatch, tmp_path):
+        )
+    # Macro must still render the input, just with an empty value.
+    macro_content = macro_path.read_text(encoding="utf-8")
+    assert 'value=""' in macro_content, (
+        "api_key_card.html macro must render inputs with an empty value= so "
+        "the user can cleanly type a new key without appending to a placeholder."
+    )
+
+
+def test_api_keys_responsive_css_has_short_viewport_sticky_rule():
+    """JTN-599: the Save button must be sticky on short laptop viewports
+    (max-height ≤ 860px covers 1280×800, 1366×768, 1280×768). Static check
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)